core/modules/views/templates/views.theme.inc

```diff
@@ -676,126 +676,115 @@ function template_preprocess_views_view_table(&$variables) {
   $columns  = $options['columns'];
-
-  $rows = array();
-  $row_indexes = array();
-
-  if ($options['alignment'] == 'horizontal') {
-    $row = array();
-    $col_count = 0;
-    $row_count = 0;
-    $count = 0;
-    foreach ($variables['rows'] as $row_index => $item) {
-      $count++;
-      $row[] = $item;
-      $row_indexes[$row_count][$col_count] = $row_index;
-      $col_count++;
-      if ($count % $columns == 0) {
-        $rows[] = $row;
-        $row = array();
-        $col_count = 0;
-        $row_count++;
-      }
-    }
-    if ($row) {
-      // Fill up the last line only if it's configured, but this is default.
-      if (!empty($handler->options['fill_single_line']) && count($rows)) {
-        for ($i = 0; $i < ($columns - $col_count); $i++) {
-          $row[] = '';
-        }
-      }
-      $rows[] = $row;
-    }
-  }
-  else {
-    $num_rows = floor(count($variables['rows']) / $columns);
-    // The remainders are the 'odd' columns that are slightly longer.
-    $remainders = count($variables['rows']) % $columns;
-    $row = 0;
-    $col = 0;
-    foreach ($variables['rows'] as $count => $item) {
-      $rows[$row][$col] = $item;
-      $row_indexes[$row][$col] = $count;
-      $row++;
-
-      if (!$remainders && $row == $num_rows) {
-        $row = 0;
-        $col++;
-      }
-      elseif ($remainders && $row == $num_rows + 1) {
-        $row = 0;
-        $col++;
-        $remainders--;
-      }
-    }
-    for ($i = 0; $i < count($rows[0]); $i++) {
-      // This should be string so that's okay :)
-      if (!isset($rows[count($rows) - 1][$i])) {
-        $rows[count($rows) - 1][$i] = '';
-      }
-    }
+  $variables['classes_array'] = array();
+  if (!$options['default_classes']) {
+    $variables['classes_array'][] = 'views-grid';
+    $variables['classes_array'][] = 'views-grid-' . $options['alignment'];
+    $variables['classes_array'][] = 'views-columns-' . $columns;
   }
+  $variables['classes_array'] = array_merge($variables['classes_array'], explode(' ', $options['wrapper_classes']));
+  $variables['classes_array'] = array_filter($variables['classes_array']);
+  // Check if there are views results to process.
+  if (!empty($variables['rows'])) {
+    $col_id = 0;
+    $row_id = 0;
+    $keys = array_keys($variables['rows']);
+    // Determine style alignment.
+    if ($options['alignment'] == 'horizontal') {
+      $rows = array();
+      $variables['row_classes'] = array();
+      foreach ($variables['rows'] as $i => $row) {
+        $classes = explode(' ', $options['column_classes']);
+        if (!$options['default_classes']) {
+          $classes[] = 'views-column-' . ($col_id + 1);
+        }
+        $rows[$row_id][$col_id] = array(
+          'classes' => implode(' ', $classes),
+          'content' => $variables['rows'][current($keys)],
+        );
+        if ($options['automatic_width']) {
+          $rows[$row_id][$col_id]['col_style'] = array('style' => 'flex: ' . (100 / $columns) . '%;');
```

Comment:
> I think that's too limiting personally. I'd think a 5 or 7-column design isnt that rare?

I use 5 all the time (see my code above, I've added it to bootstrap) but I haven't ever seen 7.

> And another thing is this is replacing established functionality. People will expect to be able to define any number of grids if theyre transferring from Drupal.

True, but they are also likely going to be confused when they realize this is using a different grid system than the rest of their site.  Long term, I'd to love to move this onto bootstrap, but it doesn't need to be right now :)